### PR TITLE
fix(next-previousTab): fixing the links in the component

### PR DESCRIPTION
**Pull request by joshjohn99**
This PR helps add better linking in the next and previous tabs.

### CODE REVIEW

**Comment by vpicone** [comment]
Hey! Thanks so much for taking this on. Custom navigation components in Gatsby can be a chore. Have you tested this using prefixed paths by chance? That's definitely bit us when trying to get this right before.

**Comment by joshjohn99** [comment]
Yes, I accounted for a href with and without a prefix.

File: packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.js [hunk 1]
```diff
@@ -106,10 +123,19 @@ const NextPreviousContainer = props => {
   });
 
   const getPreviousItem = () => {
-    const href = location.pathname.replace(pathPrefix, '');
+    // Splitting the href into a array
+    const hrefSegment = location.pathname.split('/');
```
Comment (vpicone) [inline packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.js]:
```suggestion
    const hrefSegment = location.pathname.split('/').filter(Boolean);
```

File: packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.js [hunk 2]
```diff
@@ -143,10 +169,15 @@ const NextPreviousContainer = props => {
       };
     }
 
+    // Same as the previous Tab above
+    const hrefSegment = location.pathname.split('/');
```
Comment (vpicone) [inline packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.js]: same thing here

File: packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.js [hunk 3]
```diff
@@ -143,10 +169,15 @@ const NextPreviousContainer = props => {
...
     if (nextTabItem && nextTabItem.slug) {
-      const href = location.pathname.replace(pathPrefix, '');
+      const segmentIndex = hrefSegment[hrefSegment.length - 1]
+        ? hrefSegment.length - 1
```
Comment (vpicone) [inline packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.js]: no need for these if empty string already removed

File: packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.js [hunk 4]
```diff
@@ -44,16 +49,21 @@ const getTabItems = ({ currentTitle, tabs }) => {
...
+    const slug = slugify(title, {
+      lower: true,
+    });
     return {
       title,
       slug,
-      currentTab: slug === slugify(currentTitle, { lower: true }),
+      currentTab:
+        slug ===
+        slugify(currentPage[3], {
```
Comment (vpicone) [inline packages/gatsby-theme-carbon/src/components/NextPrevious/NextPrevious.js]: Need to handle prefixed paths

**Comment by vpicone** [comment]
@joshjohn99 lmk whenever you want me to take a second look. Thanks again for working on this!

**Comment by joshjohn99** [comment]
@vpicone I added changes we talked about yesterday. Can you check it out?

**Comment by vpicone** [comment]
Path prefix was broken went ahead and pushed a fix. Our navigation page now uses the same title for two paths and works in regular and prefixed builds.

Let me know if you have any questions. If you're good to go we can merge after another review.

**Pull request merged**